hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/metadata/FlinkHoodieBackedTableMetadataWriter.java

```diff
@@ -97,17 +97,22 @@ protected void initRegistry() {
 
   @Override
   protected void commit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap) {
-    doCommit(instantTime, partitionRecordsMap, false);
+    commitInternal(instantTime, partitionRecordsMap, false, Option.empty());
+  }
+
+  @Override
+  protected List<HoodieRecord> convertRecordsToWriteClientInput(HoodieData<HoodieRecord> records) {
+    return records.collectAsList();
   }
 
   @Override
   protected void bulkCommit(String instantTime, MetadataPartitionType partitionType, HoodieData<HoodieRecord> records, int fileGroupCount) {
-    Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap = new HashMap<>();
-    partitionRecordsMap.put(partitionType, records);
-    doCommit(instantTime, partitionRecordsMap, true);
+    commitInternal(instantTime, Collections.singletonMap(partitionType, records), true, Option.empty());
   }
 
-  private void doCommit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap, boolean isInitializing) {
+  @Override
+  protected void commitInternal(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap, boolean isInitializing,
```

Comment:
@danny0405 : Can you take a look at all changes here. esply flink side of things.

Reply:
This is copied over from the existing implementation